Habitica/src/main/java/com/habitrpg/android/habitica/interactors/UseCase.java

```diff
@@ -0,0 +1,34 @@
+package com.habitrpg.android.habitica.interactors;
+
+import com.habitrpg.android.habitica.executors.PostExecutionThread;
+import com.habitrpg.android.habitica.executors.ThreadExecutor;
+
+import rx.Observable;
+
+public abstract class UseCase<Q extends UseCase.RequestValues, T> {
+
+    private final ThreadExecutor threadExecutor;
+    private final PostExecutionThread postExecutionThread;
+
+    protected UseCase(ThreadExecutor threadExecutor,
+                      PostExecutionThread postExecutionThread) {
+        this.threadExecutor = threadExecutor;
+        this.postExecutionThread = postExecutionThread;
+    }
+
+    /**
+     * Builds an {@link rx.Observable} which will be used when executing the current {@link UseCase}.
+     */
+    protected abstract Observable<T> buildUseCaseObservable(Q requestValues);
+
+    @SuppressWarnings("unchecked")
+    public Observable<T> observable(Q requestValues) {
+        return this.buildUseCaseObservable(requestValues)
+                .subscribeOn(postExecutionThread.getScheduler())
```

Comment:
I'm gonna try to get an overview of a simple UseCase and what subscribe/observeOn are called and then decide what to add and what not.

My first suggestion would be at first: let the usecase handle if it wants to use the subscribeOn or not?